### DISCUSSION TITLE
Design an implement a low level keyboard events module for GSE

### DISCUSSION THREAD

**Discussion by pzaino** [RISC-OS-Community/DME-Global-Shortcuts#3] [Announcements]
So,
after the recent events, I came to the conclusion that using closed source code is an **absolutely no go for this project**, not even if for a short time to help the end users to use the components earlier than we could possibly deliver them.

Apologies with the user-base for the extra time that this will add on the deliverables, but I strongly believe it's extremely important to maintain an enjoyable environment free on unnecessary dramas and, more importantly, avoid giving opportunities to certain subjects to use this work and effort to advertise their closed source products or formulating false allegations and try to create stories that may lead to legal actions from both sides.

RISC OS is an OS for having fun only and I hope to be able to maintain it so.

In order to ensure all the functionalities will be delivered as originally planned, I'll start to put here some ideas to write a new component for the GSE to ensure it can detect and deliver all possible keys combinations to the user-space Task to be delivered as messages to the other components in the DME as well as 3rd parties components that may wish to use this functionality.

If you are interested, then please keep an eye on the Ideas section of this forum, thanks.

**Comment by skymandr** [comment]
While not as versatile as Deep Keys, I did some tangential work for https://github.com/skymandr/TailWimpLite Most of it is in in `c.control`.

**Comment by pzaino** [comment]
Sounds like a good start to me :) and most likely we won't need many features at that level either (at least initially).

Thanks a lot!
